pkg/providers/ovirt/validation/validators/network-mapping-validator.go

```diff
@@ -0,0 +1,164 @@
+package validators
+
+import (
+	"fmt"
+
+	"kubevirt.io/client-go/kubecli"
+
+	netv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	v2vv1alpha1 "github.com/kubevirt/vm-import-operator/pkg/apis/v2v/v1alpha1"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+)
+
+//NetworkAttachmentDefinitionProvider retrieves NetworkAttachmentDefinition for given name and optional namespace
+type NetworkAttachmentDefinitionProvider interface {
+	Find(name string, namespace string) (*netv1.NetworkAttachmentDefinition, error)
+}
+
+//NetworkMappingValidator provides network mappings validation logic
+type NetworkMappingValidator struct {
+	provider NetworkAttachmentDefinitionProvider
+}
+
+//NewNetworkMappingValidator creates new NetworkMappingValidator that will use provided KubevirtClient
+func NewNetworkMappingValidator(kubevirt kubecli.KubevirtClient) NetworkMappingValidator {
+	return NetworkMappingValidator{
+		provider: &NetworkAttachmentDefinitions{
+			Client: kubevirt.NetworkClient(),
+		},
+	}
+}
+
+//ValidateNetworkMapping validates network mapping
+func (v *NetworkMappingValidator) ValidateNetworkMapping(nics []*ovirtsdk.Nic, mapping *[]v2vv1alpha1.ResourceMappingItem, crNamespace string) []ValidationFailure {
+	var failures []ValidationFailure
+	// Check whether mapping for network is required and was provided
+	if mapping == nil {
+		if v.hasAtLeastOneWithVNicProfile(nics) {
+			if _, ok := nics[0].VnicProfile(); ok {
```

Comment:
why you take here `nics[0]`?

Reply:
Good catch! It's a mistake - that check shouldn't be there at all - hasAtLeastOneWithVNicProfile does what I thought was needed.